packages/next/client/index.tsx

```diff
@@ -252,9 +253,8 @@ export async function initialize(opts: { webpackHMR?: any } = {}): Promise<{
     }
   }
 
-  if (initialData.scriptLoader) {
-    const { initScriptLoader } = require('./script')
-    initScriptLoader(initialData.scriptLoader)
+  if (initialData.scriptLoader && initialData.scriptLoader.length > 0) {
```

Comment:
I think the `if` statement should be inverted such that `if (process.env.__NEXT_SCRIPT_IMPORTED)` is checked first and then everything else is inside that.

That way dead code elimination will remove it all